src/main/java/com/exasol/cloudwatch/exasolmetrics/CompoundExasolMetricReader.java

```diff
@@ -0,0 +1,70 @@
+package com.exasol.cloudwatch.exasolmetrics;
+
+import java.sql.Connection;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * This class combines multiple {@link ExasolMetricReader}s into one.
+ */
+class CompoundExasolMetricReader implements ExasolMetricReader {
+
+    private final List<ExasolMetricReaderFactory> childReaderFactories;
+    private final Connection connection;
+    private final String schemaOverride;
+
+    /**
+     * Create a new instance of {@link CompoundExasolMetricReader}.
+     * 
+     * @param childReaderFactories factories for the child readers to combine
+     * @param connection           connection to the exasol database
+     * @param schemaOverride       if null EXA_STATISTICS is used. This parameter allows you to test this connector with
+     *                             a predefined SCHEMA instead of the unmodifiable live statistics.
+     */
+    public CompoundExasolMetricReader(final List<ExasolMetricReaderFactory> childReaderFactories,
+            final Connection connection, final String schemaOverride) {
+        this.childReaderFactories = childReaderFactories;
+        this.connection = connection;
+        this.schemaOverride = schemaOverride;
+    }
+
+    @Override
+    public List<ExasolMetricDatum> readMetrics(final Collection<String> metrics, final Instant ofMinute) {
```

Comment:
I think we should refactor this method.

You can delegate `filter` method to the factories, union results and check if all `metrics` covered.